src/Framework/TestBuilder.php

```diff
@@ -37,10 +38,11 @@ public function build(ReflectionClass $theClass, string $methodName): Test
     {
         $className = $theClass->getName();
 
-        $data = (new DataProvider)->providedData(
-            $className,
-            $methodName,
-        );
+        $data = null;
+
+        if ((new Requirements)->requirementsNotSatisfiedFor($className, $methodName) === []) {
```

Comment:
I wonder whether it would make sense to not build the whole test and e.g. return early, in case requirements will not be satisfied

atm this only prevents building the dataprovider

Reply:
I agree and it would probably solve https://github.com/sebastianbergmann/phpunit/issues/4391 as well.

Reply:
It's not possible to skip the test here, because the data provider build happens earlier when building the test suite, since a test with a data provider is a test suite itself. What this fix does is ignore the data provider and treat the test as a single test, and not a test suite.

The test will be skipped later when running the test suite. Creating a separate process happens just before running the test, but after the test suite build.